packages/stunner-editors/kie-wb-common-dmn/kie-wb-common-dmn-client/src/main/java/org/kie/workbench/common/dmn/client/editors/expressions/ExpressionEditorViewImpl.java

```
@@ -343,20 +349,34 @@ public void onEditorSelectedPageEvent(@Observes MultiPageEditorSelectedPageEvent
 
     void loadNewBoxedExpressionEditor() {
         ExpressionProps expression = ExpressionPropsFiller.buildAndFillJsInteropProp(hasExpression.getExpression(), getExpressionName(), getTypeRef());
-        String decisionNodeId = null;
+        final String decisionNodeId;
         if (hasExpression instanceof Decision) {
             decisionNodeId = ((Decision) hasExpression).getId().getValue();
         } else if (hasExpression.getExpression() instanceof FunctionDefinition) {
             decisionNodeId = getBusinessKnowledgeModel().getId().getValue();
+        } else {
+            decisionNodeId = null;
         }
-        DMNLoader.renderBoxedExpressionEditor(
-                ".kie-dmn-new-expression-editor",
-                decisionNodeId,
-                expression,
-                concat(retrieveDefaultDataTypeProps(), retrieveCustomDataTypeProps()).toArray(DataTypeProps[]::new),
-                hasExpression.isClearSupported(),
-                buildPmmlParams()
-        );
+
+        marshaller.marshall(sessionManager.getCurrentSession().getCanvasHandler().getDiagram(), new ServiceCallback<String>() {
+            @Override
+            public void onSuccess(final String item) {
+                final Object variables = DMNLoader.getVariables(item);
+                DMNLoader.renderBoxedExpressionEditor(
+                        ".kie-dmn-new-expression-editor",
+                        decisionNodeId,
+                        expression,
+                        concat(retrieveDefaultDataTypeProps(), retrieveCustomDataTypeProps()).toArray(DataTypeProps[]::new),
+                        hasExpression.isClearSupported(),
+                        buildPmmlParams(),
+                        variables);
+            }
+
+            @Override
+            public void onError(final ClientRuntimeError error) {
+
```

@danielzhe Please log an error in the console here.

---

Good catch @yesamer. We should NEVER swallow exceptions. Everything that can fail will fail. And errors needs to be properly dealt with.